<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Merge_sort</class>
 <widget class="QMainWindow" name="Merge_sort">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QLabel" name="lblTitle">
      <property name="styleSheet">
       <string notr="true">QLabel {
    font-family: &quot;Times New Roman&quot;, &quot;Segoe UI&quot;, Arial, sans-serif;
    font-size: 28px;
    font-weight: bold;
	color: qlineargradient(
        spread:pad, 
        x1:0, y1:0, x2:1, y2:0, 
        stop:0 #0077b6, stop:1 #48cae4
    );
    qproperty-alignment: AlignCenter;
    letter-spacing: 1px;
    text-transform: uppercase;
    background: transparent;
    padding: 5px;
}

</string>
      </property>
      <property name="text">
       <string>Merge Sort Visualization</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="btnReset">
        <property name="styleSheet">
         <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 18px;
    color: white;
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #0077b6,
        stop:1 #48cae4
    );
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}
</string>
        </property>
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnRandomize">
        <property name="styleSheet">
         <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 18px;
    color: white;
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #0077b6,
        stop:1 #48cae4
    );
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}
</string>
        </property>
        <property name="text">
         <string>Randomize</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnPause">
        <property name="styleSheet">
         <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 18px;
    color: white;
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #0077b6,
        stop:1 #48cae4
    );
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}
</string>
        </property>
        <property name="text">
         <string>Pause</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnStart">
        <property name="styleSheet">
         <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 18px;
    color: white;
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #0077b6,
        stop:1 #48cae4
    );
    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}
</string>
        </property>
        <property name="text">
         <string>Start</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGraphicsView" name="graphicsView"/>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QSlider" name="sliderSpeed">
        <property name="minimumSize">
         <size>
          <width>211</width>
          <height>41</height>
         </size>
        </property>
        <property name="minimum">
         <number>100</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>500</number>
        </property>
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TickPosition::TicksBelow</enum>
        </property>
        <property name="tickInterval">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="lblOperation">
        <property name="styleSheet">
         <string notr="true">QLabel {
    font-family: &quot;Times New Roman&quot;, &quot;Segoe UI&quot;, Arial, sans-serif;
    font-size: 14px;
    font-weight: bold;
	color: qlineargradient(
        spread:pad, 
        x1:0, y1:0, x2:1, y2:0, 
        stop:0 #0077b6, stop:1 #48cae4
    );
    qproperty-alignment: AlignCenter;
    letter-spacing: 1px;
    text-transform: uppercase;
    background: transparent;
    padding: 5px;
}

</string>
        </property>
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnBack">
        <property name="styleSheet">
         <string notr="true">QPushButton {
    font-family: &quot;Segoe UI Semibold&quot;, &quot;Times New Roman&quot;, Arial, sans-serif;
    font-size: 12px;
    color: white;
	background: qlineargradient(
    	spread:pad,
    	x1:0, y1:0, x2:1, y2:1,
    	stop:0 #000000,
    	stop:1 #1a1a1a
	);

    border: none;
    border-radius: 10px;
    padding: 10px 20px;
    letter-spacing: 1px;
}

QPushButton:hover {
    background: qlineargradient(
        spread:pad,
        x1:0, y1:0, x2:1, y2:1,
        stop:0 #48cae4,
        stop:1 #90e0ef
    );
}

QPushButton:pressed {
    background-color: #023e8a;
}


</string>
        </property>
        <property name="text">
         <string>back</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>18</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
